notebooks/adversarial_search_fall_2021/.ipynb_checkpoints/index-checkpoint.ipynb

```diff
@@ -0,0 +1,80 @@
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Game Theory Explanation\n",
+    "Briefly, Game Theory is designing the strategies and the steps of our player to interact in the best way, according to the rival's steps and strategies. In other words, Game theory is the study of mathematical models of strategic interactions among rational decision-makers. To know the game theory better, you can stop by <a href=\"https://en.wikipedia.org/wiki/Game_theory\">here</a>.\n",
+    "To achieve this goal, we need to express our game and its rules and actions in a mathematical form. The common model used to represent games is the model below:\n",
+    "<ul>\n",
+    "    <li>States of the game: $S_i$, starting with $S_0$</li>\n",
+    "Which means how the game will look after each action of competitors.\n",
+    "    <li>Players: P = {1,2,...,n}</li>\n",
+    "represents number of agents playing the game which varies from 1 to n.\n",
+    "    <li>Actions: $A_i$</li>\n",
+    "In every state, each player has a set of legitimate moves to change that state to another state by playing its turn.\n",
+    "    <li>Transition functions: S x A -> S</li>\n",
+    "Transition function is a function which takes a state S, and does the action A to that state, and returns another state which is built as a consequence of that action.\n",
+    "    <li>Terminal tests: S -> {True, False}</li>\n",
+    "It is a boolean function which determines whether our game has reached a final state or not.\n",
+    "    <li>Terminal utilities: S ✕ P -> R </li>\n",
+    "It takes a state S, and a player P, and returns a Real number which indicates the score of the player P (The utility of the P, to be more precise) till that moment.\n",
+    "    <li> Policy/Strategy: S -> A </li>\n",
+    "The Policy is the strategy defined by our AI program. It takes a state S, and according to the program, does action A which is the best action for that situation. \n",
+    "</ul>"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Resource Limits\n",
+    "Although the minimax algorithm would be proper for problems with relatively small state space, it isn't an efficient and feasible one for problems with more complicated and larger state space. Since the number of game states it has to examine is exponential in the depth of the tree.\n",
+    "\n",
+    "Consider a _reasonable_ chess game with $b \\approx 35$ and $m \\approx 100$ . Due to the time complexity of this algorithm which is $O(b^m)$, solving the game is not possible at all. We'll discuss some ideas to solve the problem in details below."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Depth-limited search\n",
+    "One idea might be running the algorithm up to a specified depth instead of the searching the whole tree which(_depth-limited search_).\n",
```

Comment:
review your english grammar (of example is _instead of the searching the whole tree_)